### PR TITLE
Add FLYTE_INTERNAL_POD_NAME environment variable that holds the pod name

### PR DESCRIPTION
**Pull request by bgedik**
## Tracking issue
Part of RFC https://github.com/flyteorg/flyte/blob/b3add894fd879d6b6c11e37ba292ae04bf60afeb/rfc/system/5598-deterministic-errors-distributed-training.md#flytepropeller-backend

## Why are the changes needed?
Deterministic error propagation, see: https://github.com/flyteorg/flyte/blob/b3add894fd879d6b6c11e37ba292ae04bf60afeb/rfc/system/5598-deterministic-errors-distributed-training.md#flytepropeller-backend

## What changes were proposed in this pull request?

Add FLYTE_INTERNAL_POD_NAME as a env variable to pod specs.

## How was this patch tested?

Updates to unit tests

Manual run 

image

## Check all the applicable boxes 

- [ ] I updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [X] All commits are signed-off.

## Related PRs

None

## Docs link

None

### CODE REVIEW

**Review by eapolinario — DISMISSED** [review]

**Comment by kumare3** [comment]
I would love for us to shrink all names to use _F_ format - flytekit already supports it

**Review by pingsutw — APPROVED** [review]

File: flyteplugins/go/tasks/plugins/k8s/spark/spark_test.go [hunk 1]
```diff
@@ -853,7 +853,7 @@ func TestBuildResourcePodTemplate(t *testing.T) {
 	assert.Equal(t, defaultConfig.DefaultEnvVars["foo"], findEnvVarByName(sparkApp.Spec.Driver.Env, "foo").Value)
 	assert.Equal(t, defaultConfig.DefaultEnvVars["fooEnv"], findEnvVarByName(sparkApp.Spec.Driver.Env, "fooEnv").Value)
 	assert.Equal(t, findEnvVarByName(dummyEnvVarsWithSecretRef, "SECRET"), findEnvVarByName(sparkApp.Spec.Driver.Env, "SECRET"))
-	assert.Equal(t, 9, len(sparkApp.Spec.Driver.Env))
+	assert.Equal(t, 10, len(sparkApp.Spec.Driver.Env))
```
Comment (bgedik) [inline flyteplugins/go/tasks/plugins/k8s/spark/spark_test.go:856]: @ppiegaze , @neverett  Looks like I need a review from one of you for the changes in this file. Would appreciate if you can take a quick look.

Comment (neverett) [reply flyteplugins/go/tasks/plugins/k8s/spark/spark_test.go:856]: I'll go ahead and approve, but it's strange that GitHub added us to this PR -- we're only listed as codeowners for the /docs/ directory. I'll try to fix that so we don't end up blocking future PRs.

File: flyteplugins/go/tasks/pluginmachinery/flytek8s/k8s_resource_adds.go [hunk 2]
```diff
@@ -60,6 +60,14 @@ func GetExecutionEnvVars(id pluginsCore.TaskExecutionID, consoleURL string) []v1
 			Name:  "FLYTE_INTERNAL_EXECUTION_DOMAIN",
 			Value: nodeExecutionID.Domain,
 		},
+		{
+			Name: "FLYTE_INTERNAL_POD_NAME",
```
Comment (eapolinario) [inline flyteplugins/go/tasks/pluginmachinery/flytek8s/k8s_resource_adds.go:64]: Let's rename this to `_F_PN` and leave a comment about what it points to.

**Review by eapolinario — COMMENTED** [review]
> I would love for us to shrink all names to use _F_ format - flytekit already supports it

I opened https://github.com/flyteorg/flyte/issues/5665 to track this migration to shorter names.

File: flyteplugins/go/tasks/pluginmachinery/flytek8s/k8s_resource_adds.go (hunk 2 above)
Comment (bgedik) [reply flyteplugins/go/tasks/pluginmachinery/flytek8s/k8s_resource_adds.go]: Done

File: flyteplugins/go/tasks/pluginmachinery/flytek8s/k8s_resource_adds.go [hunk 3]
```diff
@@ -60,6 +60,14 @@ func GetExecutionEnvVars(id pluginsCore.TaskExecutionID, consoleURL string) []v1
 			Name:  "FLYTE_INTERNAL_EXECUTION_DOMAIN",
 			Value: nodeExecutionID.Domain,
 		},
+		{
+			Name: "_F_PN",
```
Comment (eapolinario) [inline flyteplugins/go/tasks/pluginmachinery/flytek8s/k8s_resource_adds.go:64]: Can you a leave a comment explaining that this means pod name?

File: flytestdlib/storage/stow_store.go [hunk 4]
```diff
@@ -126,6 +126,16 @@ func (s StowMetadata) ContentMD5() string {
 	return s.contentMD5
 }
 
+type StowCursor struct {
```
Comment (eapolinario) [inline flytestdlib/storage/stow_store.go:129]: These belong to a separate change, right?

**Review by eapolinario — CHANGES_REQUESTED** [review]

**Review by eapolinario — APPROVED** [review]

**Review by eapolinario — DISMISSED** [review]

**Review by eapolinario — APPROVED** [review]

**Review by eapolinario — APPROVED** [review]
The failure is not related to your change, but good to get closer to master anyways.

**Pull request merged**